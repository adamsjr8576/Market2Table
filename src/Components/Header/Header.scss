.main-header {
  color: white;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.header-title {
  font-size: 4em;
  text-decoration: underline;
  margin: 10px 0;
}
