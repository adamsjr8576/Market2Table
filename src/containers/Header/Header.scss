.main-header {
  color: white;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.logo {
  height: 100px;
  width: 100px;
  margin-top: 10px;
  margin: 10px 0px 0px 10px;
}

.header-button {
  width: 175px;
  font-size: 1.25em;
  padding: 5px;
  border-radius: 10px;
  cursor: pointer;
  outline: 0;
  border: 0;
}

.header-button:hover,
.header-button:focus {
  transform: scale(1.05);
}

.logo-zip-header-section {
  width: 23%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.header-title {
  text-align: center;
  font-size: 4em;
  text-decoration: underline;
  margin: 10px 0;
  width: 50%;
}

.favorites-home-header-section {
  width: 23%;
}
