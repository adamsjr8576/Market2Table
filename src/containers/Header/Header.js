import React from 'react';
import { connect } from 'react-redux';
import { Route, Link } from 'react-router-dom';
import './Header.scss';
import images from '../../images/images.js';
import { removeZipCode } from '../../actions/index.js';

export const Header = ({ zipCode, favorites, removeZipCode }) => {
  const favoritesNum = favorites.length;
  return (
    <header className="main-header">
      <section className="logo-zip-header-section">
        <img src={images.m2tLogo} alt="Market 2 Table logo icon" className="logo" />
        <Route exact path="/markets" render={() => {
          return <button className="header-button" type="button" onClick={() => removeZipCode()}>Zip Code: {zipCode}</button>
        }}
        />
        <Route exact path={['/markets/:id', '/favorites']} render={() => {
          return (
            <Link to='/markets' className="header-button-link">
              <button className="header-link-button" type="button">Back</button>
            </Link>
          )
        }}
        />
      </section>
      <h1 className="header-title">Market2Table</h1>
      <section className="favorites-home-header-section">
        <Route path={['/markets', '/favorites']} render={() => {
          return (
            <Link to='/favorites' className="header-button-link">
              <button className="header-link-button">Favorites: {favoritesNum}</button>
            </Link>)
          }}
        />
        <Route path={['/markets', '/favorites']} render={() => {
          return (
            <Link to='/' className="header-button-link">
              <button className="header-link-button" onClick={() => removeZipCode()}>Home</button>
            </Link>
          )
          }}
        />
      </section>
    </header>
  )
}

export const mapStateToProps = state => ({
  zipCode: state.zipCode,
  favorites: state.favorites
});

export const mapDispatchToProps = dispatch => ({
  removeZipCode: () => dispatch( removeZipCode() )
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
