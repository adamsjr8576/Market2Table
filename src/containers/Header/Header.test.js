import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapDispatchToProps, mapStateToProps } from './Header';
import { removeZipCode } from '../../actions/index.js';

describe ('Header', () => {
  describe('Header Component', () => {
    let mockZipCode;
    let mockFavorites;
    let mockRemoveZipCode;
    let wrapper;

    beforeEach(() => {
      mockZipCode =  "80401";
      mockFavorites = [
        {
          id: "22008839",
          marketname: "Ballston Farmers Market",
          favorite: true,
          GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
          Address: "901 N Taylor St, Ballston, Virginia, 22203",
          Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
          Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
          latitude: "38.881112",
          longitude: "-77.112179"
        }
      ]
      mockRemoveZipCode = jest.fn();
      wrapper = shallow(<Header
        zipCode={mockZipCode}
        favorites={mockFavorites}
        removeZipCode={mockRemoveZipCode}
        />)
    });
    it('should match the snapshot with all correct data passed in', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('Should return an object with zipCode and favorites', () => {
      const mockMarkets = [
         {
           id: "1002192",
           marketname: "Ballston FRESHFARM Market",
           favorite: false,
           GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
           Address: "901 N Taylor St, Ballston, Virginia, 22203",
           Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
           Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
           latitude: "38.881112",
           longitude: "-77.112179"
         },
         {
           id: "22008839",
           marketname: "Ballston Farmers Market",
           favorite: false,
           GoogleLink: "http://maps.google.com/?q=38.881112%2C%20-77.112179%20(%22Ballston+FRESHFARM+Market%22)",
           Address: "901 N Taylor St, Ballston, Virginia, 22203",
           Schedule: "June - October Thursday 3:00 PM to 7:00 PM",
           Products: "Baked goods; Cheese and/or dairy products; Eggs; Fresh fruit and vegetables; Fresh and/or dried herbs; Honey; Meat; Poultry",
           latitude: "38.881112",
           longitude: "-77.112179"
         }];
      const mockState = {
        farmersMarkets: mockMarkets,
        zipCode: '80401',
        favorites: []
      }
      const expected = {
        zipCode: '80401',
        favorites: []
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('Should call dispatch with removeZipCode', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = removeZipCode();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.removeZipCode();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
