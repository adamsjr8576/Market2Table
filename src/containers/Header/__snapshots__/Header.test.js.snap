// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Component should match the snapshot with all correct data passed in 1`] = `
<header
  className="main-header"
>
  <section
    className="logo-zip-header-section"
  >
    <img
      alt="Market 2 Table logo icon"
      className="logo"
      src="m2tlogo.png"
    />
    <Route
      exact={true}
      path="/markets"
      render={[Function]}
    />
    <Route
      exact={true}
      path={
        Array [
          "/markets/:id",
          "/favorites",
        ]
      }
      render={[Function]}
    />
  </section>
  <h1
    className="header-title"
  >
    Market2Table
  </h1>
  <section
    className="favorites-home-header-section"
  >
    <Route
      path={
        Array [
          "/markets",
          "/favorites",
        ]
      }
      render={[Function]}
    />
    <Route
      path={
        Array [
          "/markets",
          "/favorites",
        ]
      }
      render={[Function]}
    />
  </section>
</header>
`;
